graphstorm-processing/docs/source/usage/amazon-sagemaker.rst

```diff
@@ -0,0 +1,141 @@
+jobs that use AWS resources.
+
+To demonstrate the usage of GSProcessing on Amazon SageMaker, we will execute the same job we used in our local
+execution example, but this time use Amazon SageMaker to provide the compute instead of our
+local machine.
+
+Upload data to S3
+-----------------
+
+Amazon SageMaker uses S3 as its storage target, so before starting
+we'll need to upload our test data to S3. To do so you will need
+to have read/write access to an S3 bucket, and the requisite AWS credentials
+and permissions.
+
+We will use the AWS CLI to upload data so make sure it is
+`installed <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
+and `configured <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-quickstart.html>`_
+in you local environment.
+
+Assuming ``graphstorm/graphstorm-processing`` is our current working
+directory we can upload the test data to S3 using:
+
+.. code-block:: bash
+
+    MY_BUCKET="enter-your-bucket-name-here"
+    REGION="bucket-region" # e.g. us-west-2
+    aws --region ${REGION} s3 sync ./tests/resources/small_heterogeneous_graph/ \
+        "${MY_BUCKET}/gsprocessing-input"
+
+.. note::
+
+    Make sure you are uploading your data to a bucket
+    that was created in the same region as the ECR image
+    you pushed in :doc:`/usage/distributed-processing-setup`.
+
+
+Launch the GSProcessing job on Amazon SageMaker
+-----------------------------------------------
+
+Once the data are uploaded to S3, we can use the Python script
+``graphstorm-processing/scripts/run_distributed_processing.py``
+to run a GSProcessing job on Amazon SageMaker.
+
+For this example we'll use a SageMaker Spark cluster with 2 ``ml.t3.xlarge`` instances
+since this is a tiny dataset. Using SageMaker you'll be able to create clusters
+of up to 20 instances, allowing you to scale your processing to massive graphs,
+using larger instances like `ml.r5.24xlarge`.
+
+Since we're now executing on AWS, we'll need access to an execution role
+for SageMaker and the ECR image URI we created in :doc:`/usage/distributed-processing-setup`.
+For instructions on how to create an execution role for SageMaker
+see the `AWS SageMaker documentation <https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-create-execution-role>`.
+
+Let's set up a small bash script that will run the parametrized processing
+job, followed by the re-partitioning job, both on SageMaker
+
+.. code-block:: bash
+
+    ACCOUNT="enter-your-account-id-here" # e.g 1234567890
+    MY_BUCKET="enter-your-bucket-name-here"
+    SAGEMAKER_ROLE_NAME="enter-your-sagemaker-execution-role-name-here"
+    REGION="bucket-region" # e.g. us-west-2
+    DATASET_S3_PATH="s3://${MY_BUCKET}/gsprocessing-input"
+    OUTPUT_BUCKET=${MY_BUCKET}
+    DATASET_NAME="small-graph"
+    CONFIG_FILE="gconstruct-config.json"
+    INSTANCE_COUNT="2"
+    INSTANCE_TYPE="ml.t3.xlarge"
+    NUM_FILES="4"
```

Comment:
Will this argument be important for users? 
If yes, we should explain how to tune its value to get better performance.
If not, could we make it transparent to users?

Reply:
It both is and it isn't, as it will determine the parallelism hint provided to Spark, but can easily be skipped. 

If users do provide it, some of the parallelism available will be limited to the number selected here.

I'll add an extra sub-section focusing on this parameter